examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,146 @@
+the limitations of a polynomial regression, both models fail when extrapolating.
 
 """
 
-import numpy as np
-import matplotlib.pyplot as plt
-from scipy import stats
-
-from sklearn.linear_model import ARDRegression, LinearRegression
-
-# %%
-# Generating simulated data with Gaussian weights
-
-# Parameters of the example
-np.random.seed(0)
-n_samples, n_features = 100, 100
-# Create Gaussian data
-X = np.random.randn(n_samples, n_features)
-# Create weights with a precision lambda_ of 4.
-lambda_ = 4.0
-w = np.zeros(n_features)
-# Only keep 10 weights of interest
-relevant_features = np.random.randint(0, n_features, 10)
-for i in relevant_features:
-    w[i] = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(lambda_))
-# Create noise with a precision alpha of 50.
-alpha_ = 50.0
-noise = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(alpha_), size=n_samples)
-# Create the target
-y = np.dot(X, w) + noise
-
 # %%
-# Fit the ARD Regression
-clf = ARDRegression(compute_score=True)
-clf.fit(X, y)
-
-ols = LinearRegression()
-ols.fit(X, y)
+# Generate synthetic dataset
+# --------------------------
+from sklearn.datasets import make_regression
+
+X, y, w = make_regression(
+    n_samples=100,
+    n_features=100,
+    n_informative=10,
+    noise=8,
+    coef=True,
+    random_state=42,
+)
 
 # %%
-# Plot the true weights, the estimated weights, the histogram of the
-# weights, and predictions with standard deviations
-plt.figure(figsize=(6, 5))
-plt.title("Weights of the model")
-plt.plot(clf.coef_, color="darkblue", linestyle="-", linewidth=2, label="ARD estimate")
-plt.plot(
-    ols.coef_, color="yellowgreen", linestyle=":", linewidth=2, label="OLS estimate"
+# Fit the regressions
+# -------------------
+import pandas as pd
+from sklearn.linear_model import ARDRegression, LinearRegression, BayesianRidge
+
+olr = LinearRegression()
+brr = BayesianRidge(compute_score=True, n_iter=30)
+ard = ARDRegression(compute_score=True, n_iter=30)
+
+olr.fit(X, y)
+brr.fit(X, y)
+ard.fit(X, y)
+
+df = pd.DataFrame(
+    {
+        "LinearRegression": olr.coef_,
+        "BayesianRidge": brr.coef_,
+        "ARDRegression": ard.coef_,
+        "true weights": w,
+    }
 )
-plt.plot(w, color="orange", linestyle="-", linewidth=2, label="Ground truth")
-plt.xlabel("Features")
-plt.ylabel("Values of the weights")
-plt.legend(loc=1)
-
-plt.figure(figsize=(6, 5))
-plt.title("Histogram of the weights")
-plt.hist(clf.coef_, bins=n_features, color="navy", log=True)
-plt.scatter(
-    clf.coef_[relevant_features],
-    np.full(len(relevant_features), 5.0),
-    color="gold",
-    marker="o",
-    label="Relevant features",
+# %%
+# Plot the true and estimated weights
+# -----------------------------------
+import matplotlib.pyplot as plt
+import seaborn as sns
+from matplotlib.colors import LogNorm
+
+plt.figure(figsize=(10, 6))
+ax = sns.heatmap(
+    df.T,
+    norm=LogNorm(),
+    cbar_kws={"label": "weight value"},
+    cmap="rocket_r",
```

Comment:
A small (personal) remainder: `ciridis` and `viridis` seem to be adapted pallets for color-blind people. 
```suggestion
    cmap="viridis_r",
```
See:
 - [Matplotlib release notes about cividis](https://matplotlib.org/stable/users/prev_whats_new/whats_new_2.2.html#cividis-colormap)
 - [those guidelines as a reference](https://www.cta-observatory.org/wp-content/uploads/2020/10/CTA_ColourBlindness_BestPractices-1.pdf)

Reply:
Thanks for the references!